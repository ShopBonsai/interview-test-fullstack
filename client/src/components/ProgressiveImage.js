import React, { useState } from 'react';
import { Media } from 'reactstrap';
import SkeletonImage from './SkeletonImage';

const ProgressiveImage = ({ src }) => {
  const [imageLoaded, setImageLoaded] = useState(false);
  return (
    <>
      {!imageLoaded && <SkeletonImage /> }
      <Media object src={src} alt="Product image cap" onLoad={() => setImageLoaded(true)} />
    </>
  )
}

export default ProgressiveImage;