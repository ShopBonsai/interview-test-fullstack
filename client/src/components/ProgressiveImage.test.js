import React from 'react';
import renderer from 'react-test-renderer';
import ProgressiveImage from './ProgressiveImage';

describe('ProgressiveImage', () => {
  it('should render correctly', () => {
    expect(
      renderer.create(
        <ProgressiveImage />
      )
    ).toMatchSnapshot();
  });
});