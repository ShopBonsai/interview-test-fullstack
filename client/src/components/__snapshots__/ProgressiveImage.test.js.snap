// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressiveImage should render correctly 1`] = `
Array [
  <div
    className="skeleton-wrapper"
  >
    <div
      className="placeholder"
    >
      <div
        className="animated-background"
      />
    </div>
  </div>,
  <img
    alt="Product image cap"
    className="media-object"
    onLoad={[Function]}
  />,
]
`;
